joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForGenDeclarationCreator.scala

```diff
@@ -75,23 +91,43 @@ trait AstForGenDeclarationCreator(implicit withSchemaValidation: ValidationMode)
     rhsParserNode: ParserNodeInfo,
     typeFullName: Option[String],
     code: String,
-    recordVar: Boolean = false
+    recordVar: Boolean = false,
+    globalStatements: Boolean = false
   ): (Ast, Ast) = {
     val rhsAst          = astForBooleanLiteral(rhsParserNode)
     val rhsTypeFullName = typeFullName.getOrElse(getTypeFullNameFromAstNode(rhsAst))
-    val localAst        = astForLocalNode(lhsParserNode, Some(rhsTypeFullName), recordVar)
-    val lhsAst          = astForNode(lhsParserNode)
-    val arguments       = lhsAst ++: rhsAst
-    val cNode = callNode(
-      rhsParserNode,
-      code,
-      Operators.assignment,
-      Operators.assignment,
-      DispatchTypes.STATIC_DISPATCH,
-      None,
-      Some(rhsTypeFullName)
+    if (globalStatements) {
+      astForGlobalVarAndConstants(rhsTypeFullName, lhsParserNode, Some(rhsAst))
+      (Ast(), Ast())
```

Comment:
Perhaps here would be a good place to have a field access-based assignment to the respective `FieldIdentifier`?

Reply:
I didn't get the question.

Reply:
If you are pointing at the modelling of the values assigned to the global variable. For that, I need to model it as a constructor method node and initialise the global member variables as part of the body. 

I will handle that part in separate PR along with respective unit tests for the same. I should have mentioned this in the TODO.